docs/source/index.mdx

```diff
@@ -0,0 +1,85 @@
+
+## Installation
+
+```bash
+pip install hffs
+```
+
+## Usage
+
+`HfFileSystem` is the cental class of the library that holds connection information and allows typical filesystem style operations like `cp`, `mv`, `ls`, `du`, `glob`, `get_file`, `put_file` etc.
+
+```python
+>>> from hffs import HfFileSystem
+>>> fs = HfFileSystem("my-username/my-dataset-repo", repo_type="dataset")
+
+>>> # List files in a directory
+>>> fs.ls("data")
+['train.csv', 'test.csv']
+
+>>> # List all ".csv" files in a repo
+>>> fs.glob("**.csv")
+['data/train.csv', 'data/test.csv']
+
+>>> # Read the contents of a remote file 
+>>> with fs.open("data/train.csv", "r") as f:
+...     train_data = f.readlines()
+
+>>> # Read all the contents of a remote file at once as a string
+>>> train_data = fs.read_text("data/train.csv")
+
+>>> # Write a remote file
+>>> with fs.open("data/validation.csv", "w") as f:
+...     f.write("text,label")
+...     f.write("Fantastic movie!,good")
+```
+
+If not explicitly provided, the `repo_type` argument defaults to `model`. Besides `model`, other supported values are `dataset` and `space`.
+
+The optional `revision` argument can be passed to open a filesystem from a specific commit (any revision such as a branch or a tag name or a commit hash).
+
+Note that, unlike the built-in `open`, the mode in `fsspec`'s `open` defaults to binary mode, i.e. "rb". This means you must explicitly set encoding as "r"/"w" in case of text mode.
+
+## Integration
+
+To use 🤗 Filesystem with a library that accepts `fsspec`'s URLs, the URL must be of the following structure:
+
+```
+hf://[<repo_type>/]<repo_id>[@<revision>]:/<path/in/repo>
+```
+
+The `repo_type` and `revision` parameters are optional. Most integrations also make it possible to pass optional parameters to the filesystem's initializer as `storage_options`, a dictionary mapping parameter names to their values.
+
+Before passing a URL to integration, you can use `fsspec.get_fs_token_paths` to ensure the URL initializes a filesystem as expected:
+
+```python
+>>> import fsspec
+
+>>> def get_filesystem_state_and_path_from_url(url):
+...     fs, _, paths = fsspec.get_fs_token_paths(url)
+...     return {"repo_type": fs.repo_type, "repo_id": fs.repo_id, "revision": fs.revision or "main", "path/in/repo": paths[0]}
+
+>>> print(get_filesystem_state_and_path_from_url("hf://dataset/johndoe/cityskapes:/images/image0.jpg"))
+{'repo_type': 'dataset', 'repo_id': 'johndoe/cityskapes', 'revision': 'main', 'path/in/repo': 'images/image0.jpg'}
+
+>>> print(get_filesystem_state_and_path_from_url("hf://johndoe/cityskapes@dev:/"))
+{'repo_type': 'model', 'repo_id': 'johndoe/cityskapes', 'revision': 'dev', 'path/in/repo': ''}
+```
+
+## Authentication
```

Comment:
Maybe also move this under Integration so users know beforehand they should be logged in with their Hugging Face account. So:

```
## Integration
## Authentication
## Usage
```